### PR TITLE
License finder

### PR DESCRIPTION
**Pull request by eguzki**
Two rake tasks added:
* License finder check `license_finder:check`
* License finder report  `license_finder:report`

License finder check added on CI

License whitelist and approval list on `/.dependency_decisions.yml`

Note: Only CSV format report

### CODE REVIEW

File: .dependency_decisions.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    :versions: []
+    :when: 2016-11-23 11:04:13.584885000 Z
+- - :whitelist
+  - 2-clause BSDL
+  - :who: Jeff Kaufmann and Richard Fontana (Red Hat Legal)
+    :why: 
+    :versions: []
+    :when: 2016-11-23 11:04:37.776089000 Z
+- - :approve
+  - colored
```
Comment (mikz) [inline .dependency_decisions.yml]: This is also approved, but why and by who? What is the license?

Comment (mikz) [reply .dependency_decisions.yml]: It does have it in the source. https://github.com/defunkt/colored/blob/master/LICENSE
Would be good to link to the source and say what license it is.

File: Rakefile [hunk 2]
```diff
@@ -17,4 +17,22 @@ rescue LoadError
...
+  desc 'Check license compliance of dependencies'
+  task :check do
+    STDOUT.puts "Checking license compliance\n"
+    unless system("license_finder --decisions-file=#{DECISION_FILE}")
+      STDERR.puts "\n*** License compliance test failed  ***\n"
+      exit 1
+    end
+  end
+
+  desc 'Generate an CSV report for licenses'
```
[the remark this answers is not in the document]
Comment (eguzki) [reply Rakefile]: not AFAIK

XML report requires [3scale XML reporter](https://github.com/3scale/license_finder_xml_reporter) which is not compatible with latest release (5.5) of license_finder. Thus, unless xml is required, is not included.

File: .dependency_decisions.yml (hunk 1 above)
Comment (eguzki) [reply .dependency_decisions.yml]: https://github.com/defunkt/colored/pull/10

almost 4 years old PR

**Pull request merged**